### PR TITLE
updated Welcome Page

### PR DESCRIPTION
**Pull request by zilto**
Updated the landing page of the Hamilton documentation.

## Changes
- Removed the comparison grid
- Shorter intro paragraph with layperson language
- Brief "Why should you use Hamilton" (this will be detailed in the user journey section)
- Architecture overview
- Who is using Hamilton figure with short companies descriptions

## How I tested this
- The docs were compiled and display properly locally
- User a spelling + grammar check

## Notes
The snippet for one image renders weirdly for local mardown previews, but is correctly compiled by the `myst` documentation parser.

### CODE REVIEW

**Comment by skrawcz** [comment]
@zilto we need the comparison grid somewhere -- thoughts?

**Comment by zilto** [comment]
> @zilto we need the comparison grid somewhere -- thoughts?

I meant to add it back in another section, but forgot to do it for this PR. I believe it could live under "Why should you use Hamilton?" or "Integrations"

**Comment by skrawcz** [comment]
> > @zilto we need the comparison grid somewhere -- thoughts?
> 
> I meant to add it back in another section, but forgot to do it for this PR. I believe it could live under "Why should you use Hamilton?" or "Integrations"

cool -- I've gotten feedback that it's something people like to see -- why not just have it at the bottom then? Since to me part of this welcome is helping position "WHERE" Hamilton is in relation to things. So either another page with a prominent link off of this one to this comparison. Or have it at the bottom?

File: docs/main.md [hunk 1]
```diff
@@ -5,32 +5,55 @@
...
+
+* **Stitch Fix** — Time series forecasting
+* **British cycling** — Telemetry analysis
+* **Joby** - Flight data processing
+* **Transfix** - Online featurization and prediction
+* **IBM** - Internal search and ML pipelines
+* **Ascena** - Feature engineering
+* **Adobe** - Prompt engineering research
+* **Axiom Cloud** - IoT data processing
+* **Oak Ridge & PNNL** - Nature project
```
Comment (skrawcz) [inline docs/main.md:53]:
```suggestion
* **Oak Ridge & PNNL** - [Naturf project](https://github.com/IMMM-SFA/naturf/tree/feature/nodes)
```

File: docs/main.md [hunk 2]
```diff
@@ -5,32 +5,55 @@
...
-| Can model GenerativeAI/LLM based workflows | ✅  |                   ❌                          |   ❌  | ❌  |  ❌   |
-| Replaces macro orchestration systems      | ❌  |                   ✅                          |   ❌  | ❌  |  ❌   |
-| Is a feature store                        | ❌  |                   ❌                          |   ✅  | ❌  |  ❌   |
-
-
-# How can you get started?
-Hop on over to our [getting started guide](get-started/index.rst)!
+**Facilitate collaboration**. By focusing on functions, Hamilton avoids sprawling code hierarchy and generates flat dataflows. Well-scoped functions make it easier to add features, complete code reviews, debug pipeline failures, and hand-off projects. Visualizations can be generated directly from your code to better understand and document it.
+
+**Reduce development time**. Hamilton dataflows are reusable across projects and context (e.g., pipeline vs. web service). The benefits of developing robust and well-tested solutions are multiplied by reusability. Off-the-shelf dataflows and boilerplates are available on the Hamilton Hub.
```
Comment (skrawcz) [inline docs/main.md:20]:
```suggestion
**Reduce development time**. Hamilton dataflows are reusable across projects and context (e.g., pipeline vs. web service). The benefits of developing robust and well-tested solutions are multiplied by reusability. Off-the-shelf dataflows are available on the [Hamilton Hub](https://hub.dagworks.io/).
```

**Review by skrawcz — APPROVED** [review]

**Pull request merged**